account_credit_control/models/credit_control_line.py

```diff
@@ -247,6 +259,15 @@ def unlink(self):
                     _('You are not allowed to delete a credit control '
                       'line that is not in draft state.')
                 )
+            if line.policy_id.auto_process_lower_levels:
+                related_lines = line.get_related_lines().filtered(
```

Comment:
it will be more performant if you add an argument to exclude self from the get_related_lines search

Reply:
You could just do something like 
`related_lines = line.get_related_lines() - line`
no ?

Reply:
I added 'exclude_ids' arg to _get_related_lines().